### PR TITLE
Feat(MicroSplit): CAREamist API

### PR DESCRIPTION
**Pull request by CatEek**
## Disclaimer

- [ ] I am an AI agent.
- [x] I have used AI and I thoroughly reviewed every line.
- [ ] I have not used AI extensively.

## Description

> [!NOTE]
> **tldr**: MicroSplit now trains and predicts end to end from `CAREamist`, and the noise model is supplied at training time instead of living in the config.

### Implementation - how did you implement the changes?

**1. Noise model generation** (same between the CAREamist and Lightning APIs):

```python
from careamics import CAREamist, NoiseModelTrainer
from careamics.config import create_n2v_config

# N2V-denoise the raw data
n2v = CAREamist(create_n2v_config(...), work_dir="nm")
n2v.train(train_data=raw)
signal, _ = n2v.predict(raw, tile_size=(256, 256))

# Fit one GMM per channel (optionally saved to .npz)
nm_trainer = NoiseModelTrainer(n_gaussian=3, n_coeff=3)
nm_trainer.train_from_pairs(signal=signal, observation=raw, axes="SCYX")
noise_model = nm_trainer.get_multichannel_model()  # -> MultiChannelNoiseModel
# nm_trainer.save("nm/")                           # -> nm/noise_model_ch{0,1}.npz
```

`CAREamist.train_noise_model(signal, observation, axes, ...)` wraps nm training: it fits, saves to `work_dir/noise_models`, attaches the result and returns it.

**2. Model training.** The config carries the likelihood *weights*, never the weights of the model itself.

```python
# 2.1 CAREamist API
config = create_microsplit_config(
    ...,
    gaussian_likelihood_weight=0.1,        # muSplit term
    noise_model_likelihood_weight=0.9,     # denoiSplit term
)
careamist = CAREamist(config)
careamist.train(
    train_data=inp, train_data_target=tgt,
    val_data=v_in, val_data_target=v_tgt,
    # ADVANCED PARAMS
    noise_model=noise_model,  
)

# 2.2 Lightning API
module = MicroSplitModule(config.algorithm_config)
module.set_noise_model(noise_model)        # object, or ["nm/ch0.npz", "nm/ch1.npz", ...]
```

HDN selects its likelihood with a `use_noise_model: bool` flag on `create_hdn_config`, replacing "did you pass a noise model"; `predict_logvar = not use_noise_model`.

**3. Inside `.train()`** — two validation points:

```python
if noise_model is not None:
    self.model.set_noise_model(noise_model)  # <-- count/channel-order validators fire HERE
datamodule = CareamicsDataModule(...)
self.trainer.fit(self.model, datamodule=datamodule)  # <-- on_fit_start validators fire HERE
```

`set_noise_model` stores the raw model and warns if the configured likelihood will not use it. `on_fit_start` raises when the noise-model likelihood is selected but nothing was injected, and rebuilds the normalized copy from the raw model on every fit.

**Guardrails.** A `dict` data config carrying MicroSplit-only fields is now rejected rather than silently downgraded to a plain `DataConfig`. Automatic validation splitting and mask-based filtering raise `NotImplementedError` for MicroSplit. A multi-channel input in the paired mode raises, since the MicroSplit input is the superimposed image and the channels to unmix belong in the target.

### New features or files

- `create_microsplit_train_val_datasets`, `create_microsplit_pred_dataset_from_data` in `dataset/factory/microsplit_factory.py`
- `set_noise_model`, `on_save_checkpoint`, `on_load_checkpoint` on `MicroSplitModule` and `HDNModule`, with shared helpers in `lightning/modules/module_utils.py`
- `CAREamist.train_noise_model`, and the `noise_model` argument of `CAREamist.train`
- `MultiChannelNoiseModel.from_npz`, `.to_config()`
- `tests/lightning/modules/test_set_noise_model.py`, `tests/lightning/data/test_datamodule_microsplit.py`

### Modified features or files

- `lightning/data/data_module.py` — MicroSplit dispatch in `setup()`,
- `lightning/modules/{microsplit,hdn}_module.py` — noise model injected rather than built from config
- `config/factories/hdn_factory.py` — `noise_model` → `use_noise_model`
- `dataset/factory/microsplit_factory.py` — single-channel input check; `_warn_unused_config_fields` compares against field defaults
- `models/lvae/noise_models.py` — `_nm_cnt` → `_nm_count`
- `tests/integration/test_microsplit_workflow.py` —  placeholder is replaced by real end-to-end tests

### Removed features or files

- `noise_model` field and `validate_noise_model_channels` from `MicroSplitAlgorithm` and `HDNAlgorithm`, plus the config-level noise-model warnings (now raised at injection time)

## How has this been tested?

- `test_set_noise_model.py` — injection from object and `.npz` paths, channel mismatch, unused-model warnings, `state_dict` exclusion, checkpoint round-trip
- `test_datamodule_microsplit.py` — MicroSplit dispatch for fit/validate/predict, dict rejection, unsupported-path errors
- `test_microsplit_factory.py` — the two bridging factories, including the train-before-convert statistics ordering
- `test_microsplit_workflow.py` — `train_noise_model`, train→predict end to end, whole-image prediction

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [ ] Documentation has been updated
- [ ] Pre-commit passes

### CODE REVIEW

**Comment by CatEek** [comment]
That's a draft, aim is to agree on the following:

**1. NM generation(shared Careamist/Lightning)**
Motivation(why is it done like that). A noise model is a frozen, pre-trained parameter of the noise_model_likelihood term of the loss — heavy trained weights, in the same category as network weights. It's produced separately (NoiseModelTrainer), from an N2V-denoised data. It is not a config value, so it never enters create_microsplit_config.
```
from careamics import CAREamist, NoiseModelTrainer
from careamics.config import create_n2v_config

# N2V-denoise the raw data to get the "signal"
n2v = CAREamist(create_n2v_config(...), work_dir="nm")
n2v.train(train_data=raw)
signal, _ = n2v.predict(raw, tile_size=(256, 256))

# Fit one GMM per channel (optionally saved to .npz)
nm_trainer = NoiseModelTrainer(n_gaussian=3, n_coeff=3)
nm_trainer.train_from_pairs(signal=signal, observation=raw, signal_axes="SCYX")
noise_model = nm_trainer.get_noise_model()     # -> MultiChannelNoiseModel (in memory)
# nm_trainer.save("nm/")                         # -> nm/noise_model_ch{0,1}.npz
```

**2.Model training**
2.1 CAREamist API
Motivation. The noise model is a training-time loss parameter derived from the same data used for training, so it goes with train(). The config doesn't carry any weights.

```
config = create_microsplit_config(
    ...,
    gaussian_likelihood_weight=0.1,        # muSplit term
    noise_model_likelihood_weight=0.9,     # denoiSplit term (the noise-model likelihood)
)

careamist = CAREamist(config)
careamist.train(
    train_data=inp, train_data_target=tgt,
    val_data=v_in, val_data_target=v_tgt,
    # ADVANCED PARAMS
    noise_model=noise_model,               # MultiChannelNoiseModel | list[str | Path]
)
```
2.2 Lightning API
```
module = MicroSplitModule(config.algorithm_config)
module.set_noise_model(noise_model)        # object, or ["nm/ch0.npz", "nm/ch1.npz", ...]
```

**3. Inside .train()**
```
def train(self, *, train_data, train_data_target=None, ...,
          noise_model: MultiChannelNoiseModel | list[str | Path] | None = None) -> None:
    ...
    if noise_model is not None:
        self.model.set_noise_model(noise_model)   # <-- count/channel-order validators fire HERE
    datamodule = CareamicsDataModule(...)
    self.trainer.fit(self.model, datamodule=datamodule)   # <-- on_fit_start validators fire HERE
```

I might've missed something, but I have more justifications if needed

**Comment by jdeschamps** [comment]
Can you comment on what's missing and what's already there? I have the feeling that this exactly what you've been working towards so far and was already discussed and agreed upon. So all good from my side.

If I remember correctly, the current configuration in `dev/microsplit_api` has a noise model configuration parameter. I think what need to discuss is whether this makes sense or not.

Run time validations of the presence of noise models are important and are already in place I believe. The point of the noise model in the configuration had to do with exporting the configuration with the complete info. How much is the noise model configuration necessary for loading a model for prediction or fine-tuning?

To be clear, I believe I was pushing for the noise model config to be saved with the checkpoints. Now I am wondering whether there would not be another way, such as saving the noise model config in the checkpoint directly but not via the configuration.

**Comment by jdeschamps** [comment]
Note that parallel work has been performed in https://github.com/CAREamics/careamics/pull/1056 and the differences should be highlighted here.

**Comment by CatEek** [comment]
> Can you comment on what's missing and what's already there? I have the feeling that this exactly what you've been working towards so far and was already discussed and agreed upon. So all good from my side.
> 
> If I remember correctly, the current configuration in `dev/microsplit_api` has a noise model configuration parameter. I think what need to discuss is whether this makes sense or not.
> 
> Run time validations of the presence of noise models are important and are already in place I believe. The point of the noise model in the configuration had to do with exporting the configuration with the complete info. How much is the noise model configuration necessary for loading a model for prediction or fine-tuning?
> 
> To be clear, I believe I was pushing for the noise model config to be saved with the checkpoints. Now I am wondering whether there would not be another way, such as saving the noise model config in the checkpoint directly but not via the configuration.

NM is not needed for inference at all. 
That's another decision, if we want to allow finetuning as a viable option, NM is good to have, still not needed bc we can fit it again in 5-10 min from params and it must be the same unless we have data/code problems. If we realize that finetuning of MS is not something ppl will do, carrying NM is pretty pointless
